src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
The intermediate state is possibly clearer from a review point of view, the big change is adding handling around `vm.GetType()` for the case that we fall through to ToggleViewModel type (i.e. respect the runtime type)